bin/nrnivmodl.in

```diff
@@ -95,214 +87,146 @@ pwd
 fi
 
-files=`echo "$files" | sed 's/\.mod//g'`
-
-if test "$files" = '*' ; then
-  files=""
-fi
+files=( "${files[@]%.mod}" )
 
 cd $MODSUBDIR
-mdir="`pwd`"
-
-  # construct file to be included by makefile to specify mod to c rule when
-  # executed in $MODSUBDIR (child folder of launch location folder)
-  MODMAKE=makemod2c_inc
-  echo "" > $MODMAKE
-  for i in $files ; do
-    case "$i" in
-      /*) f=$i;; # absolute, fine as is
-      *)  f=../$i;; # relative
-    esac
-    base_name="`basename $f`"
-    dir_name="`dirname $f`"
-    # Note: indentation for shell lines in make rules must be a tab
-    echo "\
-./$base_name.c: `escape_spaces "$f.mod"`
+mdir="$PWD"
+
+# construct file to be included by makefile to specify mod to c rule when
+# executed in $MODSUBDIR (child folder of launch location folder)
+MODMAKE=makemod2c_inc
+> "$MODMAKE"
+for i in "${files[@]}" ; do
+  case "$i" in
+    /*) f="$i";; # absolute, fine as is
+    *)  f="../$i";; # relative
+  esac
+  base_name="$(basename "$f")"
+  dir_name="$(dirname "$f")"
+  # Note: indentation for shell lines in make rules must be a tab
+  echo "\
+./${base_name// /\\ }.c: ${f// /\\ }.mod
 	@printf \" -> \$(C_GREEN)NMODL\$(C_RESET) \$<\\\n\"
-	(cd \"`unhide_spaces $dir_name`\"; MODLUNIT=\$(NRNUNITS) \$(NOCMODL) $base_name.mod -o \"$mdir\")
+	(cd \"$dir_name\"; MODLUNIT=\$(NRNUNITS) \$(NOCMODL) \"$base_name.mod\" -o \"$mdir\")
 
-./$base_name.o: ./$base_name.c
-	@printf \" -> \$(C_GREEN)Compiling\$(C_RESET) \$<\\\n\"
-	\$(COMPILE) -I\"`unhide_spaces $dir_name`\" \$(INCLUDES) @CMAKE_CXX_COMPILE_OPTIONS_PIC@ -c \$< -o \$@
-" >> $MODMAKE
-  done
-  if test "$nrnivmodl_cfiles" != "" ; then
-    # I find it very difficult to construct shell variables which are
-    # space separated lists of items containing spaces. So, for a cmake build,
-    # the nrnivmodl_cfiles variable is required to be a ';' separated list
-    # of c files which may contain spaces.
-    # Save the Internal Field Separator and interpret variables as ';'
-    # separated list of items (which may contain spaces). Restore after
-    # processing nrnivmodl_cfiles.
-    ifs="$IFS"
-    IFS=';'
-    for i in $nrnivmodl_cfiles ; do
-      i_hide=`hide_spaces "$i"`
-      case "$i_hide" in
-        /*) f=$i_hide;; # absolute, fine as is
-        *)  f=../$i_hide;; # relative
-      esac
-      base_name="`basename $f .c`"
-      dir_name="`dirname $f`"
-      echo "\
-./$base_name.o: `escape_spaces "$f"`
+./${base_name// /\\ }.o: ./${base_name// /\\ }.c
 	@printf \" -> \$(C_GREEN)Compiling\$(C_RESET) \$<\\\n\"
-	(cd \"`unhide_spaces $dir_name`\"; \$(COMPILE) \$(INCLUDES) @CMAKE_CXX_COMPILE_OPTIONS_PIC@ -c $base_name.c -o \"$mdir/$base_name.o\")
-" >> $MODMAKE
-    done
-    IFS="$ifs"
-  fi
+	\$(COMPILE) -I\"$dir_name\" \$(INCLUDES) @CMAKE_CXX_COMPILE_OPTIONS_PIC@ -c \$< -o \$@
+" >> "$MODMAKE"
+done
 
-COBJS=
 MODOBJS=
-if [ `echo "\n"` ] ; then
+if [ $(echo "\n") ] ; then
   newline="\n"
 else
   newline="\\\\n"
 fi
```

Comment:
I'm not convinced by that and I don't understand what it tried to fix. Some problem with bash & windows?